### PR TITLE
Query vulcan0x with named queries and parameters (E2D-22)

### PR DESCRIPTION
**Pull request by kkostalkowicz**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_INSTANT_ENABLED=1
```
Comment (krzkaczor) [inline .env]: do we need to commit changes like this?

File: README.md [hunk 2]
```diff
@@ -47,4 +47,16 @@ yarn cypress:dev  # to develop tests
 
 ```
 yarn cypress:run  # to run all tests
-```
\ No newline at end of file
+```
+
+## Env files
+
+By default app will try to load .env.$NODE_ENV file and fallback to .env. You can use ENV env variable to override this mechanism for example: ENV=dev will load .env.dev file.
```
Comment (krzkaczor) [inline README.md]: here we miss docs about local file and using secret to override grapgql configuration

File: src/blockchain/vulcan0x.ts [hunk 3]
```diff
@@ -38,11 +78,16 @@ export function vulcan0x(
...
-      `    ${resource}(\n${options}\n){\n` +
+      `query ${id}${queryParams} {\n` +
+      `    ${resource}(\n${options}\n) {\n` +
       `      nodes { ${fields.join(' ')} }\n` +
       '    }\n' +
-      '}'
+      '}',
+      variables: {
+        ...fromPairs(variables.map(({ name, value }) => [name, value])),
+        ...process.env.REACT_APP_GRAPHQL_DEVMODE ? { devMode: process.env.REACT_APP_GRAPHQL_DEVMODE } : {},
```
Comment (krzkaczor) [inline src/blockchain/vulcan0x.ts]: devMode is a secret? maybe it should be named `DEVMODE_SECRET`?

File: .env (hunk 1 above)
Comment (kkostalkowicz) [reply .env]: This was cherry-picked from your commit. We can revert it, but might complicate merging.

File: src/blockchain/vulcan0x.ts (hunk 3 above)
Comment (kkostalkowicz) [reply src/blockchain/vulcan0x.ts]: Should we really announce that this is a secret?

File: README.md (hunk 2 above)
Comment (kkostalkowicz) [reply README.md]: > local file
The original cherry-picked commit did not contain that information either.
> using secret to override grapgql configuration
Again, should we announce this secret?

File: .env (hunk 1 above)
Comment (krzkaczor) [reply .env]: Ah, yes nvm. I thought it's just changing line endings.

File: README.md (hunk 2 above)
Comment (krzkaczor) [reply README.md]: We shouldn't announce the secret value but the mechanism itself is public.

File: src/blockchain/vulcan0x.ts (hunk 3 above)
Comment (krzkaczor) [reply src/blockchain/vulcan0x.ts]: > We shouldn't announce the secret value but the mechanism itself is public.

File: README.md (hunk 2 above)
Comment (maciejka) [reply README.md]: Why do we want to make clear what this mechanism is about?

File: src/blockchain/vulcan0x.ts (hunk 3 above)
Comment (kkostalkowicz) [reply src/blockchain/vulcan0x.ts]: We will keep this mechanism private.

File: README.md (hunk 2 above)
Comment (kkostalkowicz) [reply README.md]: Parameter included in the docs

**Pull request merged**